go/chat/wallet/parser.go

```diff
@@ -10,21 +10,24 @@ import (
 
 var txPattern = regexp.MustCompile(
 	// Start at the beginng of the line, space, or some hand picked artisanal characters
-	`(?:^|[\s([{:;.,])` +
+	`(?:^|[\s([{:;.,!?"'])` +
 		// Have a + in front
-		`(\+` +
+		`\+` +
 		// Stellar decimal amount
 		`(\d+\.?\d*|\d*\.?\d+)` +
 		// Currency code
 		`([A-Za-z]{2,6})` +
 		// At sign and username, optional for direct messages
-		// If not used, must be followed by a non-tx-character or end of string
-		`(?:(?:@((?:[a-zA-Z0-9]+_?)+))|(?:[^A-Za-z@]|\z)))`,
+		`(?:@((?:[a-zA-Z0-9]+_?)+))?` +
+		// Sentinel group for advancing the scan
+		`()` +
+		// Must be followed by a nice character or the end of the string.
+		`(?:[\s)\]}:;.,!?"']|\z)`,
```

Comment:
maybe add a comment about \z not working inside the []s

Reply:
do we want to support `+5xlm(for lunch)`? (I'd say no, just wanted to make sure though)

Reply:
nah it's not a function